### PR TITLE
Week 1 assessment dwk1020

### PR DESCRIPTION
**Pull request by dwk1020**
# Submitting Week 1 Assessment

### ✅ Completed Files
Select the following as confirmation that the files in this repository are complete.
- [ ✅ ] Coding Challenges
- [ ✅ ] Coding Concepts
- [ ✅ ] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -20,42 +20,77 @@ const fruit4 = "kiwi"
 // --------------------1) Create a function that takes in any given number and determines if the number is below boiling point, at boiling point, or above boiling point. Boiling point is 212 degrees Fahrenheit. Use the test variables provided below. Expected output: "42 is below boiling point", "350 is above boiling point", "212 is at boiling point"
 
 // Pseudo code:
+// Create a function named "temp" that takes the boiling point and makes it an argument to return defined outcomes "is at boiling point", "is below boiling point" and "is above boiling point".  Our Boiling point = 212
```
Comment (elysemontano) [inline code-challenges.js:23]: Make sure to include your logic.  Pseudocode is meant to be a roadmap on how you plan to solve the problem.  In this case, I see conditional statements in your code, this should be included in your pseudocode as well.

File: code-challenges.js [hunk 2]
```diff
@@ -20,42 +20,77 @@ const fruit4 = "kiwi"
...
+  
+//   console.log(reverseWord("2022 Delta"));
+// Answer is 2202 atleD
 
 // --------------------4) Create a statement that will evaluate whether each number in an array is even or odd. Use the test variable provided below. Expected output: "odd" "even" "odd" "even" "odd" "even"
 
-// Pseudo code:
-
-const myArray = [13, 34, 5, 10, 27, 42]
+// Pseudo code:create array with index less than 7.
```
Comment (elysemontano) [inline code-challenges.js]: I don't see any logic in your pseudocode.  Please provide adequate pseudocode.

File: code-challenges.js [hunk 3]
```diff
@@ -20,42 +20,77 @@ const fruit4 = "kiwi"
...
 // --------------------2) Create the code that will combine the two arrays and return the length using the variables provided below. Expected output: 10
 
 // Pseudo code:
+// First must join (concat) myNum1 and myNum2 to create new array.  Console log length of new array.
+
+// const myNum1 = [1, 2, 3, 4, 5]
+// const myNum2 = [6, 7, 8, 9, 6]
+// const myNum3 = myNum1.concat(myNum2)
+// console.log(myNum3.length) 
+// Answer is 10
```
Comment (elysemontano) [inline code-challenges.js:55]: Please make sure code is uncommented out in future assessments.

File: code-challenges.js [hunk 4]
```diff
@@ -20,42 +20,77 @@ const fruit4 = "kiwi"
...
+//     }else{
+//         console.log("EVEN")
+//     }
+// }
+// Answer is ODD EVEN ODD EVEN ODD EVEN
 
 // --------------------5) Create a function that takes in a set of any two numbers and subtracts the smaller number from the larger number using the sets of test variables provided below. Expected output: 42, 3
 
-// Pseudo code:
+// Pseudo code: Console log two numbers assigned to a const that subtracts the smaller variable from the larger variable.
```
Comment (elysemontano) [inline code-challenges.js:94]: Make sure to pay attention to the prompt.  You are being asked to _create a function_ on this problem.

File: coding-concepts.js [hunk 5]
```diff
@@ -37,23 +37,24 @@ const languages = ["JavaScript", "Ruby", "Python", "C++"]
...
 // --------------------4) What will this log?
 
 const weekendDays = ["saturday", "sunday"]
-// console.log(weekendDays.toUpperCase())
+console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer:SATURDAY SUNDAY
+// b) Verify and explain:error-not a function-shouldn't be in brackets.
```
Comment (elysemontano) [inline coding-concepts.js:50]: This error occurs because .toUpperCase can only be ran on strings and not arrays.

File: coding-concepts.js [hunk 6]
```diff
@@ -37,23 +37,24 @@ const languages = ["JavaScript", "Ruby", "Python", "C++"]
...
 
 // --------------------5) What will this log?
 
 const dataTypes = ["number", "string", "Boolean"]
 // console.log(typeof dataTypes.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer:19
+// b) Verify and explain:number-length of first string in array
```
Comment (elysemontano) [inline coding-concepts.js:59]: number is returned because it is the data type of dataTypes.length which would be 3.

File: interview-questions.md [hunk 7]
```diff
@@ -15,54 +15,62 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
...
+  <!-- Researched answer:
+        Logical operators (and, or, not, and not) -->
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  <!-- Your answer:
+        Index is a position in an array that starts at 0.  An index starts with (i) and value starts with (v). (Dad joke for Elyse)  An index is giving the position to something in a paramater and the value is giving the overall quantity within the given parameter.   -->
```
Comment (elysemontano) [inline interview-questions.md:37]: Wait, wheres the dad joke you mentioned???

File: interview-questions.md [hunk 8]
```diff
@@ -15,54 +15,62 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
...
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  <!-- Your answer:Driver/Navigator.  
+        Driver is typing the code per instruction from the Navigator who is giving and proofing the typed code. -->
 
-  Researched answer:
+  <!-- Researched answer:
+        Two programmers work together at one wokstation.  The driver writes the code while the navigator reviews the written code for errors then both parties switch roles.    -->
```
Comment (elysemontano) [inline interview-questions.md:60]: While your answers in this section are accurate, they are meeting the bare minimum of the requirements for this section.  In an interview, answering with short, one liners will come off as uninterested and will not bring you success with most interviews.  I would really like to see more thorough answers next week providing as much detail and even examples when necessary.

**Review by elysemontano — CHANGES_REQUESTED** [review]
Change requested: Problem 4 in code challenges is lacking adequate pseudocode.  Please provide pseudocode that breaks down the logic shown on lines 71 - 79.  Once this is updated, I will approve this PR.
I also wanted to point out, that while I am not requiring you to change your interview answers, this section meets the bare minimum requirements and will not provide you much success in interviews if you answer with one line answers.  I would like to see more thorough answers in this section on next week's assessment.

File: code-challenges.js (hunk 1 above)
Comment (dwk1020) [reply code-challenges.js:23]: Pseudocode should show on line 23.  I am not sure why it didn't push on the original push.

**Review by dwk1020 — COMMENTED** [review]
I think u pushed the correct code in line 23.

**Review by elysemontano — APPROVED** [review]
Great work on the updates!  Approved!

**Pull request merged**